core/src/utilities/helpers/generic-helpers.js

```diff
@@ -97,18 +97,22 @@ export const getNegatedBoolString = str => {
 };
 
 export const containsAllSegments = (sourceUrl, targetPathSegments) => {
-  if (!sourceUrl || !targetPathSegments || !targetPathSegments.length) {
+  if (
+    sourceUrl === undefined ||
+    sourceUrl === null ||
+    !targetPathSegments ||
+    !targetPathSegments.length
+  ) {
     console.error(
       'Ooops, seems like the developers have misconfigured something'
     );
     return false;
   }
-
+  const mandatorySegmentsUrl = trimTrailingSlash(sourceUrl.split('?')[0]);
```

Comment:
It would be great if you could explain this part to me, maybe with a short call, because I don't understand for which case we needed to make the slicing conditional

Reply:
As discussed, it would be good to express the condition differently if possible or to leave a comment in the code to make clear why we need that